### DISCUSSION TITLE
Have Renovate install a dependency if not present?

### DISCUSSION THREAD

**Discussion by tvsbrent** [renovatebot/renovate#30201] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

I can't imagine I'm the first knucklehead to try to misuse Renovate in this way, but I can't seem to find an issue or discussion about it!

I'd like to have Renovate either update a dependency, or failing that, install the dependency when it isn't present. In this case, it would be for a Python dependency, putting it in the `requirements.txt` file if it isn't present, however I can imagine using this for a `package.json` as well.

So far my configuration attempts haven't gotten too far, although I think a custom manager may be able to do what I want? I can share more details, if needed, but I thought I'd keep this post simple and just ask if there any resources or guides on how this might be accomplished.

Thanks!

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Not supported. Can you explain what the use case is for this and when you'd use it?

**Reply by tvsbrent** [reply, marked as answer]
Definitely an odd case here, I freely admit!

The internal library I want to update is installed as part of building a Docker image that most of our projects base their project-specific image on. Those projects may install additional dependencies via a `requirements.txt` file found in their repo. A few projects already reference this internal library, to override the version, so I'd want to update them to an alpha version. However, those that don't reference it, I'd like to add that alpha version to the `requirements.txt` file.

Ultimately, I wanted to have Renovate create these branches & PRs so that the owning team's can deploy the branch to a QA environment for testing. Once we can confirm the alpha version doesn't introduce any regressions, that alpha-branch PR would be closed. I would then do a non-alpha release of the library and it would flow through Renovate normally.

However, it sounds like I'll just write my own script to do this. To sum up, my end goal here was to have Renovate manage the creation and maintenance of the PRs, with the actual underlying "upgrade" not necessarily being an upgrade.
